nss-defective-foxes/nutshell
# Cr intigration in chat

## Link to Ticket
Closes https://github.com/nss-defective-foxes/nutshell/issues/3()

## Description of Proposed Changes

-  Chat now works with initial load
- Chat can populate and adjust messages in JSON and on dom

## Steps to Test

- Once logged in, write a new message
- Make sure populates at the bottom of chat
- Make sure edit button attached only to your name and messages
- Click edit button
- Message you are trying to edit should have initial message
- Once the update button is clicked, chat should be updated with the new message replacing the old
- JSON should update the message
- When you relog-in, all information from before should be correct, including the updated message being displayed with the updated message 

```sh
git fetch --all
git checkout CRIntigrationInChat
```



## Impacted Areas in Application

List general components of the application that this PR will affect:

* Main page once logged in
* Specifically the chat section div

## Mentions @username

@AnEvilHerbivore 
@MrErin 
@wcdoster 
@sirwin4 

## Definition of Done

1. The project must be fully documented. This includes the following:
   1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
   1. Every class must be documented with purpose, author, and methods.
   1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each
